pkgs/applications/networking/flexget/default.nix

```diff
@@ -1,80 +1,48 @@
-{ lib
-, fetchFromGitHub
-, python
-, transmission
-, deluge
-, config
+{ lib, config, python
+, delugeSupport ? true, deluge ? null
 }:
 
+assert delugeSupport -> deluge != null;
```

Comment:
so unless I do an override the assertion will break? Wouldn’t it be better to set a sensitive default (no deluge support or deuge not null by default) to make a setup easier?

Reply:
No, it won't break. deluge should always be provided, unless someone explicitly does so but in that case he would have presumably disabled support. The assertion fails only when the package is called without deluge but delugeSupport is wanted. It's a common pattern used throuout nixpkgs to implement optional features.